proctor-common/src/test/java/com/indeed/proctor/common/TestForceGroupsOptionStrings.java

```diff
@@ -117,4 +133,160 @@ public void testGenerateForceGroupsString_OptionAndBucket() {
         )
                 .isEqualTo("default_to_min_live,abc1");
     }
+
+    @Test
+    public void testParseForceGroupsString_parseForcePayloadString() {
+        final Double testDouble = 0.2;
+        assertThat(ForceGroupsOptionsStrings.parseForcePayloadString("doubleValue:0.2"))
+                .isEqualTo(
+                        new Payload(testDouble)
+                );
+
+        final Double[] testDoubleArr = {0.2, 0.4, 0.6};
+        assertThat(ForceGroupsOptionsStrings.parseForcePayloadString("doubleArray:[0.2,0.4,0.6]"))
+                .isEqualTo(
+                        new Payload(testDoubleArr)
+                );
+
+        final Long testLong = 2L;
+        assertThat(ForceGroupsOptionsStrings.parseForcePayloadString("longValue:2"))
+                .isEqualTo(
+                        new Payload(testLong)
+                );
+
+        final Long[] testLongArr = {2L, 4L, 6L};
+        assertThat(ForceGroupsOptionsStrings.parseForcePayloadString("longArray:[2,4,6]"))
+                .isEqualTo(
+                        new Payload(testLongArr)
+                );
+
+        final String testString = "test,with,comma";
+        assertThat(ForceGroupsOptionsStrings.parseForcePayloadString("stringValue:\"test,with,comma\""))
+                .isEqualTo(
+                        new Payload(testString)
+                );
+
+        final String[] testStringArr = {"test1", "test2", "test3"};
+        assertThat(ForceGroupsOptionsStrings.parseForcePayloadString("stringArray:[\"test1\",\"test2\",\"test3\"]"))
+                .isEqualTo(
+                        new Payload(testStringArr)
+                );
+
+        final Map<String, Object> testMap = new HashMap<>();
+        testMap.put("testKey1", "1");
+        testMap.put("testKey2", "\\\"value2\\\"");
+        testMap.put("testKey3", "3.0");
+        testMap.put("testKey4", "[\\\"v1\\\", \\\"v2\\\"]");
+        testMap.put("testKey5", "[1.0, 2.0]");
+        testMap.put("testKey6", "1.0");
+        final Payload testP = ForceGroupsOptionsStrings.parseForcePayloadString("map:[\"testKey1\":1, \"testKey2\":\"value2\", \"testKey3\":3.0, \"testKey4\":[\"v1\", \"v2\"], \"testKey5\":[1.0, 2.0], \"testKey6\":1.0]");
+
+        assertThat(testP).isEqualTo(new Payload(testMap));
+    }
+
+    @Test
+    public void testParseForceGroupsString_GroupAndPayload() {
+        assertThat(
+                ForceGroupsOptionsStrings.generateForceGroupsString(
+                        ForceGroupsOptions.builder()
+                                .putForceGroup("abc", 1)
+                                .putForcePayload("abc", new Payload("test,with,comma"))
+                                .build())
+        )
+                .isEqualTo("abc1;{ stringValue : \"test,with,comma\" }");
```

Comment:
Can this `abc1;{ stringValue : \"test,with,comma\" }` be parsed again with `parseForceGroupsString`? Since this value will be set to cookie (You can double check by looking at `doCreateForcedGroupsCookie`), it's important to make sure the generated string is parsable.

Reply:
Good catch! Here are two approaches to implement this:

1. Create toCookieString() function that returns a payload in the format currently expected by the force payload.
2. Alter the expected format of the force payload to match to match the current toString function [(code)](https://github.com/indeedeng/proctor/blob/6445d74718ddc64a7d74dc80beaf2aca2859aeae/proctor-common/src/main/java/com/indeed/proctor/common/model/Payload.java#L171). Note: this format will create a longer string with a large amount of whitespace characters. 

I am currently implementing 1 as this is a very quick easy fix, but if option to makes more sense I will move forward with that.

Reply:
Thanks. `toCookieString()`'s implementation looks good.

I'm wondering if that implementation is better to live inside `ForceGroupsOptionsStrings` because `Payload` should not know anything about Cookie or forcePayload.

Reply:
I think that is a better location. Moved to ForceGroupOptionsStrings.java